http/cors.go

```diff
@@ -0,0 +1,32 @@
+package http
+
+import (
+	"net/http"
+
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/vault"
+)
+
+func wrapCORSHandler(h http.Handler, core *vault.Core) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		corsConf := core.CORSConfig()
```

Comment:
Just want to comment as an interested party on the work being done here.

All of this talk of which verb or what content type is a moot point.
Because Vault requires all but the authentication requests to have the `X-Vault-Token` header, *every* request will be pre-flighted, no matter what.

Note that the `Authorization` header does not appear in _any_ of the whitelisted headers lists. So, unless there's a plan to make Vault stateful using the `Cookie` header - a bad idea - supporting CORS will be required for any useful HTTP access.

[This section of the Mozilla CORS page](https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS#Preflighted_requests) explains it in detail, but basically if you have any non-standard headers (among other things), the request is preflighted.